src/KnowbaseItem.php

```diff
@@ -1736,23 +1722,63 @@ public static function getListRequest(array $params, $type = 'search')
-            'ON'  => [
-                'glpi_knowbaseitems_knowbaseitemcategories'  => 'knowbaseitems_id',
-                'glpi_knowbaseitems'             => 'id'
-            ]
-        ];
-
-        $criteria['LEFT JOIN']['glpi_knowbaseitemcategories'] = [
-            'ON'  => [
-                'glpi_knowbaseitemcategories'    => 'id',
-                'glpi_knowbaseitems_knowbaseitemcategories'  => 'knowbaseitemcategories_id'
-            ]
-        ];
-
         return $criteria;
     }
 
+    /**
+     * Clean search for Boolean FullText
+     *
+     * @since 10.0.7
+     * @param $search string
+     *
+     * @return string : cleaned search for Boolean FullText
+     **/
+    public static function computeBooleanFullTextSearch(string $search): string
+    {
+        $search_wilcard = preg_replace('/^[\p{Z}\h\v\r\n]+|[\p{Z}\h\v\r\n]+$/u', '', $search);
+        //Remove all symbols except allowed operators and space. @distance is not included, since it's unlikely a human will be using it through UI form
+        $search_wilcard = preg_replace('/[^\p{L}\p{N}_+\-<>~()*" ]/u', '', $search_wilcard);
+        //Remove all operators, that can only precede a text and that are not preceded by either beginning of string
+        $search_wilcard = preg_replace('/(?<!^|)[+\-<>~]/u', '', $search_wilcard);
+        //Remove all double quotes and asterisks, that are not preceded by either beginning of string, letter, number or space
+        $search_wilcard = preg_replace('/(?<![\p{L}\p{N}_ ]|^)[*"]/u', '', $search_wilcard);
+        //Remove all double quotes and asterisks, that are inside text
+        $search_wilcard = preg_replace('/([\p{L}\p{N}_])([*"])([\p{L}\p{N}_])/u', '', $search_wilcard);
+        //Remove all opening parenthesis which are not preceded by beginning of string
+        $search_wilcard = preg_replace('/(?<!^|)\(/u', '', $search_wilcard);
+        //Remove all closing parenthesis which are not preceded by beginning of string or are not followed by end of string
+        $search_wilcard = preg_replace('/(?<![\p{L}\p{N}_])\)|\)(?!|$)/u', '', $search_wilcard);
+        //Remove all double quotes if the count is not even
+        if (substr_count($search_wilcard, '"') % 2 !== 0) {
+            $search_wilcard = preg_replace('/"/u', '', $search_wilcard);
+        }
+        //Remove all parenthesis if count of closing does not match count of opening ones
+        if (substr_count($search_wilcard, '(') !== substr_count($search_wilcard, ')')) {
+            $search_wilcard = preg_replace('/[()]/u', '', $search_wilcard);
+        }
+        //Remove all operators, that can only precede a text and that do not have text after them (at the end of string). Do this for any possible combinations
+        $search_wilcard = preg_replace('/[+\-<>~]+$/u', '', $search_wilcard);
+        //Remove asterisk operator at the beginning of string
+        $search_wilcard = preg_replace('/^\*/u', '', $search_wilcard);
```

Comment:
Asterisks should be only located at end of a word

```suggestion
        //Remove asterisk operator not located at the end of a word
        $search_wilcard = preg_replace('/\*(?! |$)/u', '', $search_wilcard);
```